fastlane_core/lib/fastlane_core/ipa_upload_package_builder.rb

```diff
@@ -34,8 +34,8 @@ def generate(app_id: nil, ipa_path: nil, package_path: nil, platform: nil)
     private
 
     def copy_ipa(ipa_path)
-      ipa_file_name = Digest::MD5.hexdigest(ipa_path)
-      resulting_path = File.join(self.package_path, "#{ipa_file_name}.ipa")
+      ipa_file_name = File.basename(ipa_path)
```

Comment:
Thanks for the PR, I like the idea, however I think there was a reason we have the ipa file's name be a hash, to avoid conflicts. What's the reason you need the original file name on iTC? Can you use a workaround maybe?

Reply:
Yes, we create 2 types of ipas, and in Itunnes, the only way to know what kind of ipa is, is looking the fileName.
There is no other reference on iTunes related with the file, nor tag to mark this compilation or so.

I was looking and the ipa is copied on a temporary directory (with Dir.mktmpdir).
I'm check and consecutive calls to this method produce, 2 different tmp directories
Ex:
 /var/folders/p6/dh7nfy296777vf7xf37kstq80000gp/T/d20170905-14814-1i7s8ir/var/folders/p6/dh7nfy296777vf7xf37kstq80000gp/T/d20170905-14814-1ramowk

So the colission is impossible.
Anyway, if you are not sure about this, suffixing the name of ipa with that hash do the job and we still getting the original name on iTunes connect, wich is much better than nothing.
I think is better to get the original name only ( for other developers), but this could be a workaround, for us is not a problem getting a hash as suffix.

Thanks!

Reply:
I just realized, that even with this md5 digest, it's not a random one, if you have a file with the same name, and same path, the md5 will be the same.
How this could avoid a colission of 2 ipas with the same name?